content/departments/product-engineering/engineering/cloud/delivery/managed/oidc_site_admin.md

```diff
@@ -0,0 +1,103 @@
+docker exec -it sourcegraph-frontend-internal /bin/sh
+```
+
+## Initial configuration of the OpenID Connect (OIDC) Auth Provider
+
+Edit /home/sourcegraph/site-config.json, add this auth provider configuration to “auth.providers”. OpenID connect information is available in 1Password > Customer managed instances > “Okta OIDC - Managed Instance”
+
+```json
+{
+  "allowSignup": true, // this value is temporary
+  "clientID": "<value from 1password>",
+  "clientSecret": "<value from 1password>",
+  "displayName": "Sourcegraph Management",
+  "issuer": "https://sourcegraph.okta.com",
+  "requireEmailDomain": "sourcegraph.com",
+  "type": "openidconnect"
+}
+```
+
+## Restart frontend containers
+
+```
+docker ps | grep frontend | awk '{print $1}' | xargs -n1 docker restart
+```
+
+## Creating the initial management user account
+
+Log into the managed instance through the Sign In page. Click “Continue with Sourcegraph Management” to begin the authentication flow and create the initial account.
+
+## Promote initial management user to Site Admin
+
+Create a psql session into the pgsql container.
+
+```bash
+docker exec -it pg psql -U sg
+```
+
+Promote the initial account to Site Admin. Replace `<first>` and `<last>` with your first and last name respectively. Replace `<id>` using the output from the first SQL command.
+
+```sql
+// Find the initial management user account, note the id (first column)
+SELECT * FROM users WHERE username like '%<first>.<last>%';
+
+
+// Promote user to site admin, replace <id> with the
+UPDATE  users SET site_admin=’t’ WHERE id=<id>;
+```
+
+## Disable automatic account creation
+
+The initial management account should now be a site admin. Navigate to the Site Admin UI > Site Configuration and change `“allowSignup”: true` to `“allowSignup”: false` for the Sourcegraph Management auth provider. This requires other Site Admins to explicitly add users to a managed instance through the UI in order to delegate access. Save the changes.
+
+```json
+{
+  "allowSignup": false, // disable automatic creation
+  "clientID": "<value from 1password>",
+  "clientSecret": "<value from 1password>",
+  "displayName": "Sourcegraph Management",
+  "issuer": "https://sourcegraph.okta.com",
+  "requireEmailDomain": "sourcegraph.com",
+  "type": "openidconnect"
+}
+```
+
+## Apply configuration changes
+
+Restart frontend containers to apply changes.
+
+```bash
+docker ps | grep frontend | awk '{print $1}' | xargs -n1 docker restart
+```
+
+## Fix the initial account username
+
+Edit the initial account username to follow the “`sourcegraph-management-<first>-<last>`” convention. This allows the customer Site Admins to audit Sourcegraph Internal Teammate access.
+
+## Onboard Sourcegraph Teammates
+
+> **NOTE:** The convention for Sourcegraph Management usernames is `sourcegraph-management-<first>-<last>`.
+
+Add Sourcegraph Teammates as required. Use the convention for usernames and use their corresponding @sourcegraph.com email address.
```

Comment:
Do we have a better way to keep track of which SG accounts should be created? Right now the instructions are a little unclear - it sounds like we would potentially add _all_ of CS/CE.

Reply:
I linked the sources of truth. While not great, it's something :-)